portfolio/src/main/webapp/quiz.js

```diff
@@ -2,40 +2,125 @@
  */
 class Question {
-  constructor(question, answers, correct) {
+  /**
+   * @param {string} question The question that shall be posed
+   * @param {!Array<string>} answers The potential responses to the question
+   * @param {number} correctAnswerIndex The index in answers that leads to the correct response
+   */
+  constructor(question, answers, correctAnswerIndex) {
+    /**
+     * @private @const {string}
+     */
     this.question_ = question;
+    /**
+     * @private @const {Array<string>}
+     */
     this.answers_ = answers;
+    /**
+     * @private @const {number}
+     */
     this.correctAnswerIndex_ = correctAnswerIndex;
   }
 
+  /**
+   * @return {string}
+   */
   getQuestion() {
-    return question_;
+    return this.question_;
   }
 
+  /**
+   * @return {!Array<string>}
+   */
   getAnswers() {
-    return answers_;
+    return this.answers_;
   }
 
+  /**
+   * @return {number}
+   */
   getCorrectAnswerIndex() {
-    return correctAnswerIndex_;
+    return this.correctAnswerIndex_;
   }
 }
 
+const questionsArray = [];
+questionsArray.push(
+    new Question(question=`What's my GitHub username?`,
+		answers=['iolawale', 'ihsan', 'olawale314', 'ihsan314'], correctAnswerIndex=3));
+questionsArray.push(
+    new Question(question='Where do I reside?',
+		answers=['San Francisco, CA', 'Vancouver, BC', 'Toronto, ON', 'Palo Alto CA'], correctAnswerIndex=1));
+questionsArray.push(
+    new Question(question='Where do I attend university?',
+		answers=['University of British Columbia', 'Simon Fraser University',
+		  'Stanford University', 'New York University'], correctAnswerIndex=0));
+questionsArray.push(
+    new Question(question='In which of these classes did I receive the highest grade?',
+		answers=['Principles of Microeconomics', 'Honours Integral Calculus',
+		  'Basic Algorithms and Data Structures', 'Introduction to Microcomputers'], correctAnswerIndex=1));
+questionsArray.push(
+    new Question(question='In which of these classes did I receive the lowest grade?',
+		answers=['Principles of Microeconomics', 'Honours Integral Calculus',
+		  'Basic Algorithms and Data Structures', 'Introduction to Microcomputers'], correctAnswerIndex=3));
+
+/**
+ * Display the questions on the webpage, using dropdown boxes for the answers
+ */
 function presentQuiz() {
-  // create array of quiz questions 
-  const questions;
+  const questionsContainer = document.getElementById('questions-container');
+  for (let questionNum = 0; questionNum < questionsArray.length; questionNum++) {
+    const question = questionsArray[questionNum];
+    const answers = question.getAnswers();
+    const correctIndex = question.getCorrectAnswerIndex();
+    // Display the options
+    const questionStatementElement = document.createElement('p');
+    const questionStatementText = document.createTextNode(`Question ${questionNum}: ${question.getQuestion()}`);
+    questionStatementElement.appendChild(questionStatementText);
```

Comment:
This section could be rolled into a `Question` member function that returns a question in form of an element. Then you can call it like this: `const questionStatementElement = questionsArray[questionNum].getAsElement();`.

Reply:
I didn't think of this when I refactored, this would have been a fairly clean way to do it.